src/lib/eap/base.c

```diff
@@ -115,15 +115,14 @@ fr_dict_attr_autoload_t eap_base_dict_attr[] = {
 	{ NULL }
 };
 
-fr_pair_t *eap_packet_to_vp(fr_radius_packet_t *packet, eap_packet_raw_t const *eap)
+fr_pair_list_t *eap_packet_to_vp(fr_radius_packet_t *packet, eap_packet_raw_t const *eap)
 {
 	int		total, size;
 	uint8_t const *ptr;
-	fr_pair_list_t	head;
+	fr_pair_list_t	*head;
 	fr_pair_t	*vp;
-	fr_cursor_t	out;
 
-	fr_pair_list_init(&head);
+	head = fr_pair_list_alloc(packet);
```

Comment:
Yeah this seems like a bad pattern.  This function should really take a list and a ctx to do the allocations in.